README.md

```diff
@@ -31,8 +31,97 @@ Decrypted secrets from `project/config/secrets.json` (or `project/config/secrets
+
+First, move the development and test secrets to JSON secrets:
+
+```sh-session
+bin/rails runner 'Rails.root.join("config/secrets.#{Rails.env}.json").write(JSON.pretty_generate(Rails.application.secrets.to_h.without(:secret_key_base)))'
+bin/rails runner -e test 'Rails.root.join("config/secrets.#{Rails.env}.json").write(JSON.pretty_generate(Rails.application.secrets.to_h.without(:secret_key_base)))'
+```
+
+Secrets support ERB while EJSON secrets don't, so if you used that feature, you will need to move that logic in the environment configurations:
+
+**Before**:
+
+`config/secrets.yml`
+```yaml
+development:
+  some_external_service:
+    api_token: <%= ENV.fetch(SOME_EXTERNAL_SERVICE_API_TOKEN, "12345") %>
+```
+
+**After**:
+
+`config/secrets.development.json` as generated by the command above.
+```json
+{
+  "some_external_service": {
+    "api_token": "12345"
+  },
+  "something_else_entirely": "abc"
+}
+```
+
+`config/environments/development.rb`
+```ruby
+Rails.application.configure do
+  # elided
+
+  # credential should be set using []=, not the dynamic accessors
+  credentials[:some_external_service][:api_token] = ENV.fetch("SOME_EXTERNAL_SERVICE_API_TOKEN", "12345")
+
+  # top-level values must be set through `credentials.config`
+  credentials.config[:something_else_entirely] = ENV.fetch("SOMETHING_ELSE_ENTIRELY", "abc")
+end
+```
+
+You could remove the default value from the JSON or instead only set the value if the environment variable is available.
+
+Note that the code accesses the credentials as a Hash with `[]` and `[]=`. This is important because the dynamic accessor methods will set values in a different object, and credentials will behave inconsistently after that:
+
+```ruby
+Rails.application.credentials.some_external_service.api_token = "foo"
+Rails.application.credentials[:some_external_service][:api_token] # => "12345"
+```
+
+Also note the code sets top-level values through `credentials.config`, because `credentials#[]=(key, value)` sets values in a different object.
+
+```ruby
+Rails.application.credentials[:something_else_entirely] = "foo"
+Rails.application.credentials[:something_else_entirely] # => "abc"
+```
+
+Make sure there's no code using the dynamic accessors before setting the configuration in the Hash, or the values won't be accessible from the dynamic accessor:
+
+```ruby
+Rails.application.credentials.something_else_entirely # just accessing is enough to cause the issue
+Rails.application.credentials[:some_external_service][:api_token] = "foo"
+Rails.application.credentials.some_external_service.api_token # => "12345"
+```
+
+You're now ready to use credentials instead of secrets:
+
+```sh-session
+git ls-files | xargs ruby -pi -e 'gsub("Rails.application.secrets", "Rails.application.credentials")' --
+```
+
+Once you're running Rails 7.1, to avoid the deprecation warning from the use of secrets in `ejson-rails`, you can require another file from your Gemfile:
```

Comment:
```suggestion
To avoid the deprecation warning from the use of secrets in `ejson-rails` once you're running Rails 7.1, require another file from your Gemfile:
```